src/cudadev/DataFormats/SoAView.h

```diff
@@ -0,0 +1,375 @@
+}
+
+#define generate_SoA_const_view(CLASS, STORES_LIST, VALUE_LIST)                                                                     \
+struct CLASS {                                                                                                                      \
+                                                                                                                                    \
+  /* these could be moved to an external type trait to free up the symbol names */                                                  \
+  using self_type = CLASS;                                                                                                          \
+                                                                                                                                    \
+  /**                                                                                                                               \
+   * Helper/friend class allowing SoA introspection.                                                                                \
+   */                                                                                                                               \
+  struct SoAMetadata {                                                                                                              \
+    friend CLASS;                                                                                                                   \
+    /* Alias store types to name-derived identifyer to allow simpler definitions */                                                 \
+    _ITERATE_ON_ALL(_DECLARE_VIEW_STORE_TYPE_ALIAS, ~, STORES_LIST)                                                                 \
+                                                                                                                                    \
+    /* Alias member types to name-derived identifyer to allow simpler definitions */                                                \
+    _ITERATE_ON_ALL(_DECLARE_VIEW_MEMBER_TYPE_ALIAS, const, VALUE_LIST)                                                             \
+  private:                                                                                                                          \
+    SOA_HOST_DEVICE_INLINE SoAMetadata(const CLASS& parent): parent_(parent) {}                                                     \
+    const CLASS& parent_;                                                                                                           \
+  };                                                                                                                                \
+  friend SoAMetadata  ;                                                                                                             \
+  SOA_HOST_DEVICE_INLINE const SoAMetadata soaMetadata() const { return  SoAMetadata(*this); }                                      \
```

Comment:
Confusing `const`
```suggestion
  SOA_HOST_DEVICE_INLINE SoAMetadata soaMetadata() const { return  SoAMetadata(*this); }                                      \
```

Reply:
Fixed in a8bf1ff.

Reply:
This is the same use case as https://github.com/cms-patatrack/pixeltrack-standalone/pull/211#discussion_r768066785.